OpenRA.Mods.Common/Effects/Missile.cs

```diff
@@ -662,8 +662,13 @@ int IncreaseAltitude(int predClfDist, int diffClfMslHgt, int relTarHorDist, int
 				{
 					// Aim to attain cruise altitude as soon as possible while having the absolute value
 					// of vertical facing bound by the maximum vertical rate of turn
+					var dat = args.SourceActor.World.Map.DistanceAboveTerrain(pos);
 					var vDist = new WVec(-diffClfMslHgt - info.CruiseAltitude.Length, -speed, 0);
 					desiredVFacing = (sbyte)vDist.HorizontalLengthSquared != 0 ? vDist.Yaw.Facing : vFacing;
+
+					if (dat.Length > info.CruiseAltitude.Length)
```

Comment:
@pchote I'm not 100% sure given the missile code as a whole is quite a bit above me, but checking against `-diffClfMslHgt` instead of `dat.Length` indeed fixes the bug just as well on flat terrain and probably better on TS/RA2 cliffs (though I didn't notice any practical problems with TS cliffs while using `dat.Length`, either).